drivers/axi_core/axi_dmac/axi_dmac.c

```diff
@@ -410,18 +410,20 @@ int32_t axi_dmac_transfer_start(struct axi_dmac *dmac,
 		}
 	}
 
-	/* Cyclic transfers set to HW or SW depending on size for MEM to DEV. */
+	/* Clear the DMA_CYCLIC flag for all transfers */
+	axi_dmac_read(dmac, AXI_DMAC_REG_FLAGS, &reg_val);
```

Comment:
Do you need to clear this flag even if the transfer is cyclic (and the flag would be set again after this)? If not, you can write this as:

```
if (cyclic) {
    set flag
} else {
    clear flag
}
```

Also, even though it's not in the scope of this PR, maybe it's worth to refactor the cyclic condition in the `if` bellow, which would make the code cleaner. I would create a `can_offload_cyclic` (or a similar name) variable which would be set if those conditions are met.

Reply:
Yes, there are two ways of performing a cyclic transfer: hardware or software. This complicates the if condition.

There is also the hw_cyclic property of the DMA that indicates if hardware cyclic transfers are supported.